### ISSUE TITLE
GUI — Display Issues Upon Font Style & Size Changes

### ISSUE THREAD

**Issue by touwys** [Borewit/listFix#149]
When certain font style and size combinations are changed from the programme default settings, some of the GUI elements adjust improperly. Examples:

![GUI font style behaviour](https://user-images.githubusercontent.com/65730626/229285223-e516754a-6498-43a2-9b1a-d9684a3796c9.jpg)

![image](https://user-images.githubusercontent.com/65730626/229285288-5c15ade9-e04b-4928-bf6b-7c93062b1143.png)

![image](https://user-images.githubusercontent.com/65730626/229285301-5948fb0a-6d94-4fff-a188-b29b650565d5.png)

![image](https://user-images.githubusercontent.com/65730626/229285380-7f5eba09-eefe-4cf5-ba53-581677d3e0c4.png)

___

**Comment by Borewit** [comment]
I don't think it practically possible to prevent the larger font sizes from exceeding the available space on GUI components.

Neither I expect putting further constraints will be be a good solution, as fonts scale differently.

We could remove this feature.

I don't want to put a lot of effort in the GUI, as it may a better investment to migrate the current Swing GUI to [JavaFX](https://openjfx.io/).

**Comment by Borewit** [comment]
> As I continue to test it intermittently, should I inclusively post comments, observations, or issues — all under ["Issues"](https://github.com/Borewit/listFix/issues)? Proposal: It could be fruitful to open a separate subject that takes care of the general discussion around current listFix() releases.

Ideally issues are requests for change, bug requests, feature request, improvement. Clearly scoped change. Discussions in issues should be primarly about that issue. Same goes for the PR, which is about a change, which could resolve 1 or more issues. 
 
I have actually never used, but we could use [discussions](https://github.com/Borewit/listFix/discussions) for dicussions beyond the scope of issue's and PR's. But that is actually what it is designed for. For other projects I used Discord, which offers a bit more privacy.

> Have you finished picking all the "low-hanging fruit" now, or are there more to come? If you would, please see whether you can fit into your plans, https://github.com/Borewit/listFix/issues/116#issue-1600016422: It's a bother to identify a matched file — especially so if it's to be found quickly somewhere among numerous other entries — once its location is established.

It is not the end of the world, but that comment does not belong here, it belongs in #116. If we discuss it here, it will be hard to follow for others, we create confusing links from this issue to that that issue etc

**Comment by touwys** [comment]
> We could remove this feature.

> I don't want to put a lot of effort in the GUI, as it may a better investment to migrate the current Swing GUI to [JavaFX](https://openjfx.io/).

It is a tough call. It appears to be the more prudent option right now, especially if you're certain that you're going to migrate the Swing GUI to JavaFX.

**Comment by touwys** [comment]
Just for interest's sake: Another anomaly I have noticed about changing the font style, is that once the font style of **certain** fonts is changed, it does not change the playlist text inside the editor accordingly. So that may be one more good reason to remove the option. Example:

**[Changing the font style](https://user-images.githubusercontent.com/65730626/229539573-7b6ec120-2326-4c2b-8827-c575c84288d6.png)**

**Comment by touwys** [comment]
For some reason, this release: [Build listFix_2.7.0-PR156-1](https://github.com/Borewit/listFix/pull/156#issuecomment-1494703547) appears to have solved this issue — to a large degree. The GUI-elements did scale nicely with the default font, as well as one other that I have tried (Candara_14). Theme applied: Nimbus

_Examples:_

![image](https://user-images.githubusercontent.com/65730626/229717985-ee0030ea-5bcf-4907-b3d5-0ca5d10c02ce.png)

![image](https://user-images.githubusercontent.com/65730626/229718092-a89dd2c7-a2f0-4b54-976c-1bcbd34a8b03.png)

As previously reported, and again in only in one instance observed (applying the default font sans serif_14), the size of the text of a playlist that is already open inside the editor, did not change according to size set.

I propose that we keep the Font Style Options — if only temporarily, while working towards another solution. The little inconvenience of scaling everything, is made up for in the easier reading of the larger fonts and text spread.

**Comment by Borewit** [comment]
> For some reason, this release: https://github.com/Borewit/listFix/pull/156#issuecomment-1494703547 appears to have solved this issue — to a large degree.

That should be the result of #152, which removed the buggy jgoodies look & feel themes. On changing the look & feel (maybe also when changing font size), some update event was send to all visual components. One the the jgoodies  threw a NullPointerException somewhere in that process which prevented the update event reaching most components, which prevent them from adapting to that change or even crash / froze the app as you experienced yourself.

Let's close this issue for now. Feel free to re-open if required.
